nomad/datasource_nomad_job_test.go

```diff
@@ -0,0 +1,119 @@
+package nomad
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccNomadJob_Basic(t *testing.T) {
+	var testJob api.Job
+	jobId := acctest.RandString(8)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckNomadJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNomadJobConfig_basic(jobId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNomadJobExists("nomad_job.foobar", &testJob),
+				),
+			},
+			{
+				Config: testAccCheckNomadJobConfig_basic(jobId),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.nomad_job.foobar", "id", fmt.Sprintf("%s", jobId)),
```

Comment:
This could probably be combined with the first TestStep, and ideally would check more of the computed attributes to make sure they're set correctly.